File: getting-started/todo/java/core/src/main/java/com/amplifyframework/samples/core/ListActivity.java
package com.amplifyframework.samples.core;

import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.amplifyframework.samples.core.databinding.ActivityMainBinding;

public abstract class ListActivity extends AppCompatActivity {
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = ActivityMainBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        binding.floatingActionButton.setOnClickListener(v -> fabAction());
    }

    public abstract void fabAction();
}

File: getting-started/todo/core/src/main/java/com/amplifyframework/samples/core/ListActivity.kt
package com.amplifyframework.samples.core

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity


abstract class ListActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val fab: View = findViewById(R.id.floatingActionButton)
        fab.setOnClickListener {
            fabAction()
        }
    }

    abstract fun fabAction()
}
